browser-extension/mv2/src/client/js/ruleExecutionHandler.js

```diff
@@ -141,3 +181,48 @@ RQ.RuleExecutionHandler.notifyRuleAppliedToWidget = (ruleId) => {
     );
   }
 };
+
+RQ.RuleExecutionHandler.notifyRuleAppliedToImplicitWidget = (rule) => {
+  const implicitTestRuleWidget = document.querySelector("rq-implicit-test-rule-widget");
+  implicitTestRuleWidget.style.display = "block";
+
+  if (implicitTestRuleWidget) {
+    implicitTestRuleWidget.setAttribute("applied-rule-id", rule.id);
+    implicitTestRuleWidget.setAttribute("applied-rule-name", rule.name);
+    implicitTestRuleWidget.setAttribute("applied-rule-type", rule.ruleType);
+    implicitTestRuleWidget.dispatchEvent(
+      new CustomEvent("new-rule-applied", {
+        detail: {
+          appliedRuleId: rule.id,
+          appliedRuleName: rule.name,
+          appliedRuleType: rule.ruleType,
+        },
+      })
+    );
+  }
+};
+
+const getImplicitTestRuleWidgetConfig = async () => {
+  await chrome.storage.local.get(RQ.IMPLICIT_RULE_TESTING_WIDGET_CONFIG, function (result) {
+    RQ.RuleExecutionHandler.implictTestRuleWidgetConfig = result[RQ.IMPLICIT_RULE_TESTING_WIDGET_CONFIG];
+  });
+};
+
+const checkAppliedRuleAndNotifyWidget = (rule) => {
+  let notifyWidget = true;
+  const implicitTestRuleConfig = RQ.RuleExecutionHandler.implictTestRuleWidgetConfig;
+
+  if (implicitTestRuleConfig.visibility === RQ.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.OFF) {
+    notifyWidget = false;
+    return;
+  }
+
+  if (implicitTestRuleConfig.visibility === RQ.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.SPECIFIC) {
+    if (!implicitTestRuleConfig.ruleTypes.includes(rule.ruleType)) {
+      notifyWidget = false;
+      return;
+    }
+  }
+
+  if (notifyWidget) RQ.RuleExecutionHandler.notifyRuleAppliedToImplicitWidget(rule);
+};
```

Comment:
Why do we need the variable `notifyWidget`?

Can we write it like this?
```suggestion
const checkAppliedRuleAndNotifyWidget = (rule) => {

  const implicitTestRuleConfig = RQ.RuleExecutionHandler.implictTestRuleWidgetConfig;

  if (implicitTestRuleConfig.visibility === RQ.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.OFF) {
    return;
  }

  if (implicitTestRuleConfig.visibility === RQ.IMPLICIT_RULE_TESTING_WIDGET_VISIBILITY.SPECIFIC) {
    if (!implicitTestRuleConfig.ruleTypes.includes(rule.ruleType)) {
      return;
    }
  }

RQ.RuleExecutionHandler.notifyRuleAppliedToImplicitWidget(rule);
};
```